src/lib/index/publish.js

```diff
@@ -27,25 +27,77 @@
 'use strict';
 
 const
-	Amqp = require('./shared/amqp'),
+	_ = require('lodash'),
+	amqp = require('amqplib'),
+	{ validate } = require('./shared/configValidator'),
 
 	EventEmitter = require('events'),
 
 	ERROR_EVENT = 'error_event',
 
 	emitter = new EventEmitter(),
 
-	send = ({ eventName, buffer, config }) => {
-		return Amqp.apply(channel => channel.sendToQueue(eventName, buffer), config)
-			.catch(err => {
-				emitter.emit(ERROR_EVENT, err.message);
-				throw err;
-			});
-	};
+	publishers = new Map();
+
+class Publisher {
+
+	constructor(config) {
+		validate(config);
+		this.config = _.cloneDeep(config);
+	}
+
+	async init() {
+		this.connection = await amqp.connect(this.config.cloudamqpUrl);
+		this.channel = await this.connection.createChannel();
+	}
+
+	async publish(queue, buffer) {
+		return await this.channel.sendToQueue(queue, buffer);
+	}
+
+	async close() {
+		await this.connection.close();
+	}
+
+	static async newPublisher(config) {
+		const publisher = new Publisher(config);
+		await publisher.init();
+		return publisher;
+	}
+
+}
+
+async function send({ eventName, buffer, config }) {
+
+	try {
+		let publisherPromise = publishers.get(config),
+			publisher = null;
+
+		if (!publisherPromise) {
+			publisherPromise = Publisher.newPublisher(config);
+			publishers.set(config, publisherPromise);
+		}
+
+		publisher = await publisherPromise;
+		return await publisher.publish(eventName, buffer);
```

Comment:
Equivalent but less code: `return publisher.publish(eventName, buffer)`

Reply:
`return await blabla` is also generally less efficient, as it unwraps the Promise result and then wraps it up again.

Reply:
True

Reply:
Not equivalent if you want to log errors in this function...

Reply:
Touché